File: solutions/java/0977.java
class Solution {
  public int[] sortedSquares(int[] A) {
    int n = A.length;
    int l = 0;
    int r = n - 1;
    int[] ans = new int[n];

    while (l <= r)
      ans[--n] = Math.abs(A[l]) > Math.abs(A[r]) ? A[l] * A[l++] : A[r] * A[r--];

    return ans;
  }
}

File: solutions/cpp/0977.cpp
class Solution {
 public:
  vector<int> sortedSquares(vector<int>& A) {
    vector<int> ans(A.size());

    int i = A.size() - 1;
    for (int l = 0, r = A.size() - 1; l <= r;)
      if (abs(A[l]) > abs(A[r]))
        ans[i--] = A[l] * A[l++];
      else
        ans[i--] = A[r] * A[r--];

    return ans;
  }
};